nearcache_map.go

```diff
@@ -319,3 +440,88 @@ func (ncm *nearCacheMap) handleBatchInvalidationMsg(rth *inearcache.RepairingHan
 	})
 	return rth.HandleBatch(keys, sources, partitions, seqs)
 }
+
+func (ncm *nearCacheMap) populateResultFromNearCache(keys []interface{}, entries []types.Entry) ([]interface{}, error) {
+	// see: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#populateResultFromNearCache
+	var missKeys []interface{}
+	var i int
+	for _, k := range keys {
+		nk, err := ncm.toNearCacheKey(k)
+		if err != nil {
+			return nil, err
+		}
+		cached, ok, err := ncm.getCachedValue(nk, true)
+		if err != nil {
+			return nil, err
+		}
+		if ok && cached != nil {
```

Comment:
Can `cached` be nil? Java docs say you cannot cache a nil value. I may be wrong

Reply:
Yes, `cached` can be `nil`, but the user cannot cache a nil value. The code corresponds to this reference implementation code at com/hazelcast/client/map/impl/nearcache/NearCachedClientMapProxy.java:431
```
if (cached != null && cached != NOT_CACHED) {
```